dpar/src/features/input_layers.rs

```diff
@@ -28,33 +28,13 @@ impl AddressedValues {
     ///
     /// Multiple addresses are used to e.g. address the left/rightmost
     /// dependency of a token on the stack or buffer.
-    pub fn from_buf_read<R>(read: R) -> Result<Self>
+    pub fn from_buf_read<R>(mut read: R) -> Result<Self>
     where
         R: BufRead,
     {
-        let mut addrs = Vec::new();
-
-        for line in read.lines() {
-            let line = line?;
-
-            if line.trim().is_empty() {
-                continue;
-            }
-
-            let mut layer_addrs = parse_addressed_value_templates(line.as_bytes())?;
-
-            if layer_addrs.len() != 1 {
-                bail!(ErrorKind::ConfigError(format!(
-                    "Combinatory features are not \
-                     supported/necessary: {}",
-                    line
-                )));
-            }
-
-            addrs.push(layer_addrs.remove(0));
-        }
-
-        Ok(AddressedValues(addrs))
+        let mut data = String::new();
+        read.read_to_string(&mut data)?;
```

Comment:
That would work here as well, just pass in your line. Of course, you'll get back a `Vec`, but I think it's more correct, since someone could pass in an empty line.

Note also that in the current grammar (in contrast to the previous grammar), a feature specification can also be split up in multiple lines.

Reply:
I can't add much to this because I simply don't know how these files are actually generated. In any way the impact of parsing line by line vs. reading the entire file should be negligible in the scale of the program so the current version alllows for a bit more flexibility